### PR TITLE
ddl, test: integrate add index with the distribute framework

### LINKED ISSUE #42974: integrate `add index` with the framework

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #42974

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -1864,6 +1870,79 @@ func (w *worker) addTableIndex(t table.Table, reorgInfo *reorgInfo) error {
...
+			return errors.Errorf("cannot find global task with ID %d", globalTask.ID)
+		}
+
+		if found.State == proto.TaskStateSucceed {
+			return nil
+		}
+
+		if found.State == proto.TaskStateFailed || found.State == proto.TaskStateCanceled || found.State == proto.TaskStateReverted {
+			return errors.Errorf("ddl task stopped with state %s", found.State)
+		}
```
Comment (tangenta) [inline ddl/index.go:1942]: If the task fails, returning the source error instead of "ddl task stopped..." is easier to understand.

**Review by lcwangchao — APPROVED** [review]

File: ddl/index.go (hunk 1 above)
Comment (tangenta) [reply ddl/index.go:1944]: OK.

**Pull request merged**